src/parquet/file_reader.cc

```diff
@@ -0,0 +1,764 @@
+#include "compression.hh"
+
+#include <seastar/parquet/file_reader.hh>
+#include <seastar/parquet/exception.hh>
+#include <seastar/parquet/overloaded.hh>
+
+#include <seastar/core/file.hh>
+#include <seastar/core/seastar.hh>
+
+#include <thrift/protocol/TCompactProtocol.h>
+#include <thrift/transport/TBufferTransports.h>
+
+#include <iostream>
+#include <sstream>
+#include <limits>
+
+namespace parquet {
+
+////////////////////////////////////////////////////////////////////////////////
+// thrift deserialization
+
+namespace {
+
+template <typename DeserializedType>
+void deserialize_thrift_msg(
+        const uint8_t serialized_msg[],
+        uint32_t* serialized_len,
+        DeserializedType& deserialized_msg
+) {
+    using ThriftBuffer = apache::thrift::transport::TMemoryBuffer;
+    auto tmem_transport = std::make_shared<ThriftBuffer>(const_cast<uint8_t*>(serialized_msg), *serialized_len);
```

Comment:
This line is a cause of failure for me when I compile with Thrift 0.14, because it tries to free `serialized_msg` in the destructor... no idea why. It works when I artificially allocate memory here and copy, but that's not how it *should* work at all.

That's quite frustrating, since the stock Thrift for Fedora is 0.10 and it requires using `boost::shared_ptr` instead of `std::shared_ptr`, which makes `parquet/` compilation impossible, and Thrift versions 0.11-0.13 produce compilation errors themselves, so I can't easily build and install them... Btw, we need to add a CMake line that ensures Thrift is at least 0.11, otherwise `parquet/` does not compile due to not being able to assing std::shared_ptr to boost::shared_ptr.

I'll debug that tomorrow, or try to find a Thrift version that works (unless it's somehow an error in this patch, but I assume you'd notice that the test always fails with a segfault).